src/App.tsx

```diff
@@ -14,11 +15,107 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isShown: boolean,
+  isReversed: boolean,
+  sortBy: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    isShown: false,
+    isReversed: false,
+    sortBy: 'id',
+  };
+
+  showAll = () => {
+    this.setState({ isShown: true });
+  };
+
+  reverse = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  reset = () => {
+    this.setState({ sortBy: 'id', isReversed: false });
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  render() {
+    const { isShown, isReversed, sortBy } = this.state;
+    const visibleGoods = [...goodsFromServer];
+
+    visibleGoods.sort((a: string, b: string): number => {
+      switch (sortBy) {
+        case 'alphabet': return a.localeCompare(b);
+        case 'length': return a.length - b.length;
+        default: return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="App content">
+        <h1 className="title mt-6">Goods</h1>
+
+        {!isShown && (
```

Comment:
ternary added. could you, please, explain the logic behind adding buttons and methods to the GoodsList component?
because in the video Misha used it to render a list of friends only.

Reply:
You should just move them there, and pass `goodsFrom Server` in props, logic of list should be in it's component